pkg/kubelet/cm/topologymanager/topology_manager_test.go

```diff
@@ -0,0 +1,664 @@
+							Preferred:      true,
+						},
+						{
+							SocketAffinity: NewTestSocketMask(0, 1),
+							Preferred:      false,
+						},
+					},
+				},
+			},
+			expected: TopologyHint{
+				SocketAffinity: NewTestSocketMask(1),
+				Preferred:      true,
+			},
+		},
+	}
+
+	for _, tc := range tcases {
+		mngr := manager{}
+		mngr.hintProviders = tc.hp
+		actual := mngr.calculateAffinity(v1.Pod{}, v1.Container{})
+		if !actual.SocketAffinity.IsEqual(tc.expected.SocketAffinity) {
+			t.Errorf("Expected SocketAffinity in result to be %v, got %v", tc.expected.SocketAffinity, actual.SocketAffinity)
+		}
+		if actual.Preferred != tc.expected.Preferred {
+			t.Errorf("Expected Affinity preference in result to be %v, got %v", tc.expected.Preferred, actual.Preferred)
+		}
+	}
+}
+
+func TestAddContainer(t *testing.T) {
+	testCases := []struct {
+		name        string
+		containerID string
+		podUID      types.UID
+	}{
+		{
+			name:        "Case1",
+			containerID: "nginx",
+			podUID:      "0aafa4c4-38e8-11e9-bcb1-a4bf01040474",
+		},
+		{
+			name:        "Case2",
+			containerID: "Busy_Box",
+			podUID:      "b3ee37fc-39a5-11e9-bcb1-a4bf01040474",
+		},
+	}
+	mngr := manager{}
+	mngr.podMap = make(map[string]string)
+	for _, tc := range testCases {
+		pod := v1.Pod{}
+		pod.UID = tc.podUID
+		err := mngr.AddContainer(&pod, tc.containerID)
+		if err != nil {
+			t.Errorf("Expected error to be nil but got: %v", err)
+		}
+		if val, ok := mngr.podMap[tc.containerID]; ok {
+			if reflect.DeepEqual(val, pod.UID) {
+				t.Errorf("Error occurred")
+			}
+		} else {
+			t.Errorf("Error occurred, Pod not added to podMap")
+		}
+	}
+}
+
+func TestRemoveContainer(t *testing.T) {
+	testCases := []struct {
+		name        string
+		containerID string
+		podUID      types.UID
+	}{
+		{
+			name:        "Case1",
+			containerID: "nginx",
+			podUID:      "0aafa4c4-38e8-11e9-bcb1-a4bf01040474",
+		},
+		{
+			name:        "Case2",
+			containerID: "Busy_Box",
+			podUID:      "b3ee37fc-39a5-11e9-bcb1-a4bf01040474",
+		},
+	}
+	var len1, len2 int
+	mngr := manager{}
+	mngr.podMap = make(map[string]string)
+	for _, tc := range testCases {
+		mngr.podMap[tc.containerID] = string(tc.podUID)
+		len1 = len(mngr.podMap)
+		err := mngr.RemoveContainer(tc.containerID)
+		len2 = len(mngr.podMap)
+		if err != nil {
+			t.Errorf("Expected error to be nil but got: %v", err)
+		}
+		if len1-len2 != 1 {
+			t.Errorf("Remove Pod resulted in error")
+		}
+	}
+
+}
+func TestAddHintProvider(t *testing.T) {
+	var len1 int
+	tcases := []struct {
+		name string
+		hp   []HintProvider
+	}{
+		{
+			name: "Add HintProvider",
+			hp: []HintProvider{
+				&mockHintProvider{},
+			},
+		},
+	}
+	mngr := manager{}
+	for _, tc := range tcases {
+		mngr.hintProviders = []HintProvider{}
+		len1 = len(mngr.hintProviders)
+		mngr.AddHintProvider(tc.hp[0])
+	}
+	len2 := len(mngr.hintProviders)
+	if len2-len1 != 1 {
+		t.Errorf("error")
+	}
+}
+
+func TestAdmit(t *testing.T) {
```

Comment:
Please add some test cases that exercise pod admission failing due to strict policy when no preferred alignment found, and those same scenarios succeeding with preferred or none policy.